File: library-appcompat/src/main/java/com/cz/android/sample/library/appcompat/processor/FragmentClassActionProcessor.java
package com.cz.android.sample.library.appcompat.processor;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;

import com.cz.android.sample.api.item.RegisterItem;
import com.cz.android.sample.library.appcompat.SampleFragmentContainerActivity;
import com.cz.android.sample.processor.clazz.ClassActionProcessor;

/**
 * @author Created by cz
 * @date 2020-01-28 20:48
 * @email bingo110@126.com
 */
public class FragmentClassActionProcessor extends ClassActionProcessor {

    @Override
    public boolean isInstance(Class item) {
        return super.isInstance(item)&& Fragment.class.isAssignableFrom(item);
    }

    @Override
    public void run(FragmentActivity context, RegisterItem registerItem, Class item) throws Exception {
        SampleFragmentContainerActivity.startActivity(context,registerItem);
    }
}

File: core/src/main/java/com/github/jackchen/android/core/processor/FragmentClassActionProcessor.kt
package com.github.jackchen.android.core.processor

import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import com.github.jackchen.android.core.appcompat.SampleFragmentContainerActivity
import com.github.jackchen.android.core.exception.SampleFailedException
import com.github.jackchen.android.sample.api.SampleItem

/**
 * @author JackChen
 */
class FragmentClassActionProcessor : ActionProcessor {

  override fun isAvailable(clazz: Class<*>): Boolean {
    return Fragment::class.java.isAssignableFrom(clazz)
  }

  @Throws(SampleFailedException::class)
  override fun execute(context: AppCompatActivity, sampleItem: SampleItem) {
    SampleFragmentContainerActivity.startActivity(context, sampleItem)
  }
}
